bio_hansel/scripts/main.py

```diff
@@ -0,0 +1,146 @@
+    parser = argparse.ArgumentParser(
+        prog=SCRIPT_NAME, formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument(
+        '-i',
+        '--input-vcf',
+        required=True,
+        help=
+        'vcf file containing the list of SNVs that are related to the genomes')
+
+    parser.add_argument(
+        '-g',
+        '--reference-genome-file',
+        required=True,
+        help='Path to reference genome name, can be .gb or .gbk format')
+    parser.add_argument(
+        '-o',
+        '--output-folder-name',
+        required=True,
+        help=
+        'Output file name for the genomes, default folder name is /usr/home/genomes'
+    )
+    parser.add_argument(
+        '-v',
+        '--verbose',
+        action='count',
+        default=0,
+        help=
+        'Logging verbosity level (-v == show warnings; -vvv == show debug info)'
+    )
+
+    parser.add_argument(
+        '-n',
+        '--schema-name',
+        help=
+        'A unique name for the schema file that is generated, the default is just'
+        ' {bio_hansel-schema-reference_genome_name}-{schema_version}'
+    )
+
+    parser.add_argument(
+        '-m',
+        '--schema-version',
+        help=
+        'An optional version number for the schema file that is generated'
+
+    )
+
+    parser.add_argument(
+        '-t',
+        '--minimum-threshold',
+        required=True,
+        type=float,
+        help=
+        'Minimum threshold to be tested using the hierarchical clustering scheme'
+    )
+
+    parser.add_argument(
+        '-u',
+        '--maximum-threshold',
+        required=True,
+        type=float,
+        help=
+        'Maximum threshold to be tested using the hierarchical clustering scheme'
+    )
+
+    return parser
+
+
+def main():
+    home_folder = os.path.expanduser('~')
+    parser = init_parser()
+    if len(sys.argv[1:]) == 0:
+        parser.print_help()
+        parser.exit()
+    args = parser.parse_args()
+    output_folder_name = args.output_folder_name
+    min_threshold = args.minimum_threshold
+    max_threshold = args.maximum_threshold
+
+    if (min_threshold >= max_threshold):
+        logging.error("max_threshold has to be bigger than min_threshold")
+    init_console_logger(3)
+    vcf_file = args.input_vcf
+    reference_genome_path = args.reference_genome_file
+
+    reference_genome_name = reference_genome_path.split("/")[-1]
+    reference_genome_name=reference_genome_name.split(".")[-2]
+    print(reference_genome_name)
+    logging.info('using genbank file from %s', reference_genome_path)
+
+    if args.schema_version is not None:
+        schema_version = args.schema_version
+    else:
+        schema_version = "0.1.0"
+
+    if args.schema_name is not None:
+        schema_name = args.schema_name - {schema_version}
+    else:
+        schema_name = f"bio_hansel-schema-{reference_genome_name}-{schema_version}"
+
+    output_directory = f"{home_folder}/{output_folder_name}"
```

Comment:
It's never a good idea to join paths with a delimiter that only will work on one platform. Please use `os.path.join(dir, filename)` instead.

Reply:
I ended up letting the user inputting their own path, and the command line right now just takes in a string that represents the path instead, addressed in commit 4277dffbf5dc652244a1425057a3cee485b4b4e1